src/Core/Grid/Definition/Factory/CustomerAddressGridDefinitionFactory.php

```diff
@@ -54,7 +54,24 @@ final class CustomerAddressGridDefinitionFactory extends AbstractGridDefinitionF
     public function __construct(HookDispatcherInterface $hookDispatcher, ?Request $currentRequest)
     {
         parent::__construct($hookDispatcher);
-        $this->backUrl = $currentRequest ? $currentRequest->getUri() : '';
+
+        // Remove useless query parameters to avoid some strange behaviors with delete workflow
+        // This way, behavior of "customer address grid" is consistent with `AddressGridDefinitionFactory`
+        // (should be solved using proper front-end and API architecture)
```

Comment:
I think this comment can be removed. It is a comment good to add inside the PR description because it helps reviewers, but it has no value to stay inside the software for maybe 20 years 😄  .

Comments inside the code should help developers understand what the code does and why, not give hints about how it could be solved in another architecture.

Reply:
Not to mention "proper" or "not proper" is a very open question 😄

Reply:
Hmmm ... I agree for most of useless comments i did.

But i think it is important to precise somewhere that "customer address grid" and "AddressGridDefinitionFactory" behaviors have to be consistent.
As we are on a refactoring plan that could make problem simpler, I think it is not a good idea to try to develop a central/generic code to implement both behaviors and ensure code and product consistency.

But how could we ensure that a coming development / fix won't break this product consistency ?

:thinking:  Thinking this way my comment looks a bit naive. I should at least ad the same in "AddressGridDefinitionFactory" ^^' ...

I got this link for product identity and style guide : https://build.prestashop.com/prestashop-ui-kit/?path=/story/list-group--list-group
Do you have any document for behavior description ?

Reply:
> But i think it is important to precise somewhere that "customer address grid" and "AddressGridDefinitionFactory" behaviors have to be consistent.

I agree. We can keep the comment that mentions we need this consistency. I suggest to remove the "this could be solved by a proper architecture" part as it will not be useful for the reader of this class I think.